src/WebJobs.Script.WebHost/BreakingChangeAnalysis/ChangeAnalysisService.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+using System.Threading;
+using System.Threading.Tasks;
+using DotNetTI.BreakingChangeAnalysis;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public sealed class ChangeAnalysisService : IHostedService, IBreakingChangeAnalysisService, IDisposable
+    {
+        private readonly IEnvironment _environment;
+        private readonly IPrimaryHostStateProvider _hostStateProvider;
+        private readonly IChangeAnalysisStateProvider _changeAnalysisStateProvider;
+        private readonly ILogger _logger;
+        private readonly SemaphoreSlim _analysisSemaphore;
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private Task _analysisTask;
+        private bool _disposed = false;
+
+        public ChangeAnalysisService(ILoggerFactory loggerFactory,
+                                     IEnvironment environment,
+                                     IChangeAnalysisStateProvider changeAnalysisStateProvider,
+                                     IPrimaryHostStateProvider hostStateProvider)
+        {
+            _environment = environment;
+            _hostStateProvider = hostStateProvider;
+            _changeAnalysisStateProvider = changeAnalysisStateProvider;
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryBreakChangeAnalysis);
+            _analysisSemaphore = new SemaphoreSlim(1, 1);
+            _cancellationTokenSource = new CancellationTokenSource();
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            if (!_environment.IsPlaceholderModeEnabled() && !_environment.IsCoreTools())
```

Comment:
Use `IOptionsMonitor<StandbyOptions>` here instead.

Reply:
Any reason for this to be `IOptionsMonitor` here? This is a Job Host scoped service, `IOptions` should be sufficient, right?

Reply:
Ah yes -- it should be. Just trying to prevent going directly against _environment.

Reply:
Discussed offline. But keeping the `_environment` use here because of issues with the setup for the standby.